core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -414,6 +414,16 @@ public PlanWithProperties visitTopN(TopNNode node, PreferredProperties preferred
                     break;
                 case PARTIAL:
                     child = planChild(node, PreferredProperties.any());
+                    // If source is pre-sorted, partial topN can be replaced with partial Limit N
+                    List<LocalProperty<Symbol>> desiredProperties = node.getOrderingScheme().getOrderBy().stream()
```

Comment:
This most likely deserves a separate rule but such rule would have to derive properties too (which is not the rule way). I think it's fine to keep simplification here since it's rather trivial and all information are at hand. cc @martint